src/response/into_response.rs

```diff
@@ -69,23 +69,23 @@ impl IntoResponse for &'_ str {
 //     }
 // }
 
-// impl<T: IntoResponse, U: IntoResponse> IntoResponse for Result<T, U> {
-//     fn into_response(self) -> Response {
-//         match self {
-//             Ok(r) => r.into_response(),
-//             Err(r) => {
-//                 let res = r.into_response();
-//                 if res.status().is_success() {
-//                     panic!(
-//                         "Attempted to yield error response with success code {:?}",
-//                         res.status()
-//                     )
-//                 }
-//                 res
-//             }
-//         }
-//     }
-// }
+impl<T: IntoResponse, U: IntoResponse> IntoResponse for Result<T, U> {
+    fn into_response(self) -> Response {
+        match self {
+            Ok(r) => r.into_response(),
+            Err(r) => {
+                let res = r.into_response();
+                if res.status().is_success() {
```

Comment:
this seems like an unnecessary limitation, maybe the user wants to return a different kind of success response on certain errors, so I wouldn’t force this, especially not with a panic

Reply:
I am conflicted about this - I think it's semantically confusing to return something that will map to a successful status code in the `Err` variant of a `Result`.
Should it be a `panic`? I guess that's up for discussion.